api/lib/domain/services/certification-challenges-service.js

```diff
@@ -16,46 +16,33 @@ module.exports = {
   async pickCertificationChallenges(placementProfile) {
     const knowledgeElementsByCompetence = await knowledgeElementRepository
       .findUniqByUserIdGroupedByCompetenceId({ userId: placementProfile.userId, limitDate: placementProfile.profileDate });
-
     const knowledgeElements = KnowledgeElement.findDirectlyValidatedFromGroups(knowledgeElementsByCompetence);
     const answerIds = _.map(knowledgeElements, 'answerId');
 
-    const challengeIdsCorrectlyAnswered = await answerRepository.findChallengeIdsFromAnswerIds(answerIds);
-
-    const allChallenges = await challengeRepository.findFrenchFranceOperative();
-    const challengesAlreadyAnswered = challengeIdsCorrectlyAnswered.map((challengeId) => Challenge.findById(allChallenges, challengeId));
-
-    challengesAlreadyAnswered.forEach((challenge) => {
-      if (!challenge) {
-        return;
-      }
-
-      const userCompetence = _getUserCompetenceByChallengeCompetenceId(placementProfile.userCompetences, challenge);
-
-      if (!userCompetence || !userCompetence.isCertifiable()) {
-        return;
-      }
-
-      challenge.skills
-        .filter((skill) => _skillHasAtLeastOneChallenge(skill, allChallenges))
-        .forEach((publishedSkill) => userCompetence.addSkill(publishedSkill));
-    });
+    const alreadyAnsweredChallengeIds = (await answerRepository.findChallengeIdsFromAnswerIds(answerIds))
+      .map((id) => { return { id }; });
 
+    const allFrFrOperativeChallenges = await challengeRepository.findFrenchFranceOperative();
     const userCompetences = UserCompetence.orderSkillsOfCompetenceByDifficulty(placementProfile.userCompetences);
     let certificationChallengesByCompetence = {};
 
     userCompetences.forEach((userCompetence) => {
+      if (!userCompetence.isCertifiable()) {
+        return;
+      }
```

Comment:
@laura-bergoens on pourrait faire ca?
```suggestion
    userCompetences
        .filter(userCompetence => userCompetence.isCertifiable())
        .forEach((userCompetence) => {
```

Reply:
Oui, c'est fait dans un commit après